### ISSUE TITLE
Enabling more compiler warnings

### ISSUE THREAD

**Issue by baentsch** [open-quantum-safe/liboqs#1083]
After reading https://queue.acm.org/detail.cfm?id=3468263 one might be tempted to activate more compiler warning flags than just `-Wall -Wextra`). When doing this in `liboqs` (only adding `-Wnull-dereference` and `-Wduplicated-branches`) the following code "shortcomings" are unearthed:

```
[26/1202] Building C object src/sig/picnic/CMakeFiles/picnic.dir/external/picnic_impl.c.o
../src/sig/picnic/external/picnic_impl.c: In function ‘oqs_sig_picnic_impl_sign’:
../src/sig/picnic/external/picnic_impl.c:1060:3: warning: potential null pointer dereference [-Wnull-dereference]
   generate_seeds(pp, context, prf->round[0].seeds[0], prf->salt);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[83/1202] Building C object src/kem/sike/CMakeFiles/sike.dir/external/P610/P610_compressed.c.o
In file included from ../src/kem/sike/external/P610/P610_compressed.c:407:0:
../src/kem/sike/external/P610/../ec_isogeny.c: In function ‘Ladder3pt_dual’:
../src/kem/sike/external/P610/../ec_isogeny.c:690:8: warning: this condition has identical branches [-Wduplicated-branches]
     if (AliceOrBob == ALICE) {
        ^
```
-> Might it be worth while notifying upstream code maintainers of this and if fixed, also activating these warnings when building `liboqs`? Another idea may be to activate `-fanalyzer` when moving to gcc10/11. Any other flags worth while adding to make the code even more robust?

**Comment by baentsch** [comment]
Not receiving comments probably means this is not exactly relevant. Keeping open for some time, but labeling "future work"

**Comment by thomwiggers** [comment]
re: `-fanalyzer`: I've experimented with it in https://github.com/PQClean/PQClean/pull/449 but it has problems with the current version of the SPHINCS+ implementation in PQClean/liboqs. Once https://github.com/PQClean/PQClean/pull/470 is merged I think the implementations in PQClean all pass under `-fanalyzer` (and I want to rebase that PR and undraft it then)
